lib/vanagon/driver.rb

```diff
@@ -92,7 +92,7 @@ def run
       puts e.backtrace.join("\n")
       raise e
     ensure
-      if @engine == "hardware"
+      if @engine.name == "hardware"
```

Comment:
This could've been @engine.is_a? Vanagon::Engine::Hardware, or a new method teardown_on_failure that no-ops for everything except Hardware. Adding a redundant name makes me sad.